lxd/daemon.go

```diff
@@ -1100,6 +1117,88 @@ func (d *Daemon) PasswordCheck(password string) error {
 	return nil
 }
 
+func (d *Daemon) ExpireLogs() error {
+	entries, err := ioutil.ReadDir(shared.LogPath())
+	if err != nil {
+		return err
+	}
+
+	result, err := dbContainersList(d.db, cTypeRegular)
+	if err != nil {
+		return err
+	}
+
+	newestFile := func(path string, dir os.FileInfo) time.Time {
+		newest := dir.ModTime()
+
+		entries, err := ioutil.ReadDir(path)
+		if err != nil {
+			return newest
+		}
+
+		for _, entry := range entries {
+			if entry.ModTime().After(newest) {
+				newest = entry.ModTime()
+			}
+		}
+
+		return newest
+	}
+
+	for _, entry := range entries {
+		// Check if the container still exists
+		if shared.StringInSlice(entry.Name(), result) {
+			// Remove any log file which wasn't modified in the past 48 hours
```

Comment:
I know some users are asking for this, but I think a real policy like this needs to be clearly documented and announced somewhere.  I could easily see running a test and coming back to the results a week later, expecting the logs to still be there...  I'm not excited about yet more config options, and the policy here seems as good as any...  i wonder if 'lxc log --help' would be a place to documen this.  (also like the idea of 'lxc log <containername>' as a nicer way of asking for 'lxc info --show-log <containername>'.  Then maybe 'lxc log --set-expiry 0 <containername>".  Probably not, but...